### PR TITLE
[PLAT-773] django-health-check improvements

### PR DESCRIPTION
**Pull request by tony-zebra**
FYI, I managed to touch a lot more code than was strictly necessary to address a few things I didn't like.  These include:
- Making class names across the health checks more consistent
- Simplifying how health checks of the same type (e.g., db, celery) are registered with the plugin manager
- Making the health checks more self contained (i.e., passing in configuration settings instead of calling out to get them)

### CODE REVIEW

File: health_check/plugins.py [hunk 1]
```diff
@@ -1,67 +1,47 @@
...
     def __init__(self):
-        self._registry = []  # type: List[Tuple[Type[BaseHealthCheckBackend], dict]]
+        self._registry = []  # type: List[Tuple[str, Type[BaseHealthCheckBackend], Tuple, dict]]
 
     def reset(self):
         """Reset (clear out) the registry, e.g. for testing purposes."""
         self._registry.clear()
 
-    def register(self, plugin: Type[BaseHealthCheckBackend], **options):
+    def register(self, app_name: str, plugin: Type[BaseHealthCheckBackend], args: tuple = (), kwargs: dict = {}):
```
Comment (drmaples) [inline health_check/plugins.py]: be careful with mutable default args as they may end up causing subtle bugs

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

File: health_check/views.py [hunk 2]
```diff
@@ -23,56 +31,40 @@ def get(self, request, *args, **kwargs):
...
+        data = {'components': component_results}
+        return JsonResponse(data, status=get_response_code(plugins))
 
 
 class StatusView(View):
     def get(self, request):
-        return HttpResponse('OK', status=200)
+        required_plugins = plugin_manager.execute_plugins(required_only=True)
+        status = get_response_code(required_plugins)
+        msg = 'OK' if status == 200 else 'SICK'
```
Comment (drmaples) [inline health_check/views.py]: 🤒

File: health_check/views.py [hunk 3]
```diff
@@ -23,56 +31,40 @@ def get(self, request, *args, **kwargs):
...
-    def is_unknown(value):
-        return value is None or value == VersionView.UNKNOWN
+    def get_info() -> (Dict[str, str], int):
+        # purposely not using default argument with os.getenv() so we can test this code better
+        data: Dict[str, str] = dict(
+            repo=os.getenv('CI_REPO_NAME') or VersionView.UNKNOWN,
+            version=os.getenv('CI_COMMIT_ID') or VersionView.UNKNOWN,
+            branch=os.getenv('CI_BRANCH') or VersionView.UNKNOWN
+        )
+        status = 503 \
```
Comment (drmaples) [inline health_check/views.py]: nit: using `\` is unnecessary and can be replaced with `()` which is significantly easier on the 👀. however, in this case a 1 liner should suffice.

```python
status = (
    503
    if VersionView.UNKNOWN in data.values()
    else 200
)
```

File: health_check/celery/backends.py [hunk 4]
```diff
@@ -1,26 +1,18 @@
...
-
-    @staticmethod
-    def factory() -> List[BaseHealthCheckBackend]:
-        return [CeleryHealthCheck(queue) for queue in current_app.amqp.queues]
-
-    def __init__(self, queue):
-        super(CeleryHealthCheck, self).__init__()
+class CeleryHealthCheckBackend(BaseHealthCheckBackend):
+    def __init__(self, queue, timeout=2):
+        super(CeleryHealthCheckBackend, self).__init__()
```
[the remark this answers is not in the document]
Comment (tony-zebra) [reply health_check/celery/backends.py]: I'm going to make this python 2 compatible, so will leave as-is for now.  But thanks, I didn't know about the zero arg syntax.

[the remark this answers is not in the document]
Comment (drmaples) [reply health_check/celery/backends.py]: there is lots of stuff in here already that wont work with py2:
- f-strings
- type annotations
- maybe the relative imports

**Pull request merged**